src/OpenFeature.Contrib.Providers.Flagd/Cache.cs

```diff
@@ -0,0 +1,170 @@
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+
+[assembly: InternalsVisibleTo("DynamicProxyGenAssembly2")]
+
+namespace OpenFeature.Contrib.Providers.Flagd
+{
+    internal interface ICache<TKey, TValue>
+    {
+        void Add(TKey key, TValue value);
+        TValue TryGet(TKey key);
+        void Delete(TKey key);
+        void Purge();
+    }
+    class LRUCache<TKey, TValue> : ICache<TKey, TValue> where TValue : class
+    {
+        private readonly int _capacity;
```

Comment:
A c# int is a signed 32 bit, so max value here is `2,147,483,647`. If each of those entries is 10 bytes (probably a reasonable size for a flag-resolution) that's about ~200MB, which seems a bit small. I think perhaps this should be a long?

TBH I think that we might need the same change in Java. What do you think?